api/transport/handler_panic_test.go

```diff
@@ -0,0 +1,151 @@
+		panic(msg)
+	}
+
+	err := transport.DispatchUnaryHandler(
+		context.Background(),
+		transport.UnaryHandlerFunc(handler),
+		time.Now(),
+		&transport.Request{},
+		nil,
+	)
+	expectMsg := fmt.Sprintf("panic: %s", msg)
+	assert.Equal(t, err.Error(), expectMsg)
+}
+
+func TestDispatchOnewayHandlerWithPanic(t *testing.T) {
+	msg := "I'm panicking in a oneway handler!"
+	handler := func(context.Context, *transport.Request) error {
+		panic(msg)
+	}
+
+	err := transport.DispatchOnewayHandler(
+		context.Background(),
+		transport.OnewayHandlerFunc(handler),
+		&transport.Request{},
+	)
+	expectMsg := fmt.Sprintf("panic: %s", msg)
+	assert.Equal(t, err.Error(), expectMsg)
+}
+
+func TestDispatchStreamHandlerWithPanic(t *testing.T) {
+	msg := "I'm panicking in a stream handler!"
+
+	handler := func(_ *transport.ServerStream) error {
+		panic(msg)
+	}
+
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockStream := transporttest.NewMockStream(mockCtrl)
+	mockStream.EXPECT().Request().Return(
+		&transport.StreamRequest{
+			Meta: &transport.RequestMeta{},
+		}).Times(1)
+	mockServerStream, _ := transport.NewServerStream(mockStream)
+	err := transport.DispatchStreamHandler(
+		transport.StreamHandlerFunc(handler),
+		mockServerStream,
+	)
+	expectMsg := fmt.Sprintf("panic: %s", msg)
+	assert.Equal(t, err.Error(), expectMsg)
+}
+
+func TestInvokeUnaryHandlerWithPanic(t *testing.T) {
+	msg := "I'm panicking in a unary handler!"
+	handler := func(context.Context, *transport.Request, transport.ResponseWriter) error {
+		panic(msg)
+	}
+
+	err := transport.InvokeUnaryHandler(
+		transport.UnaryInvokeRequest{
+			Context:   context.Background(),
+			StartTime: time.Now(),
+			Request:   &transport.Request{},
+			Handler:   transport.UnaryHandlerFunc(handler),
+			Options:   transport.NewInvokerOptions(transport.Logger(zap.NewNop())),
+		},
+	)
+	expectMsg := fmt.Sprintf("panic: %s", msg)
+	assert.Equal(t, err.Error(), expectMsg)
+}
+
+func TestInvokeOnewayHandlerWithPanic(t *testing.T) {
+	msg := "I'm panicking in a oneway handler!"
+	handler := func(context.Context, *transport.Request) error {
+		panic(msg)
+	}
+
+	err := transport.InvokeOnewayHandler(transport.OnewayInvokeRequest{
+		Context: context.Background(),
+		Request: &transport.Request{},
+		Handler: transport.OnewayHandlerFunc(handler),
+		Options: transport.NewInvokerOptions(transport.Logger(zap.NewNop())),
+	},
+	)
+	expectMsg := fmt.Sprintf("panic: %s", msg)
+	assert.Equal(t, err.Error(), expectMsg)
+}
+
+func TestInvokeStreamHandlerWithPanic(t *testing.T) {
+	msg := "I'm panicking in a stream handler!"
+
+	handler := func(_ *transport.ServerStream) error {
+		panic(msg)
+	}
+
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockStream := transporttest.NewMockStream(mockCtrl)
+	mockStream.EXPECT().Request().Return(
+		&transport.StreamRequest{
+			Meta: &transport.RequestMeta{},
+		}).Times(1)
+	mockServerStream, _ := transport.NewServerStream(mockStream)
```

Comment:
similar, let's add the error check since `transport.NewServerStream` could return an error, even if we're passing in a mock.